algorithms/unit_tests/TestStdAlgorithmsConstraints.cpp

```diff
@@ -81,5 +81,122 @@ TEST(std_algorithms, is_admissible_to_std_algorithms) {
+  EXPECT_DEATH(
+      { KE::Impl::expect_no_overlap(first_d0, last_d0, first_d1, last_d1); },
+      "Kokkos contract violation:.*");
+
+  auto strided_view_1d_0 = Kokkos::subview(strided_view_1d, range0);
+  auto strided_view_1d_1 = Kokkos::subview(strided_view_1d, range1);
+  auto first_st0         = KE::begin(strided_view_1d_0);  // [0, 12]
+  auto last_st0          = first_st0 + strided_view_1d_0.extent(0);
+  auto first_st1         = KE::begin(strided_view_1d_1);  // [3, 15]
+  auto last_st1          = first_st1 + strided_view_1d_1.extent(1);
+  // Does not overlap since offset (=3) is not divisible by stride (=2)
+  KE::Impl::expect_no_overlap(first_st0, last_st0, first_st1, last_st1);
+
+  // Iterating over the same range without overlapping
+  Kokkos::View<value_type[2][extent0], Kokkos::LayoutLeft> static_view_2d{
+      "std-algo-test-2d-contiguous-view-static"};
+  auto sub_static_view_1d_0 = Kokkos::subview(static_view_2d, 0, Kokkos::ALL);
+  auto sub_static_view_1d_1 = Kokkos::subview(static_view_2d, 1, Kokkos::ALL);
+  auto sub_first_s0         = KE::begin(sub_static_view_1d_0);  // 0, 2, 4, ...
+  auto sub_last_s0          = sub_first_s0 + sub_static_view_1d_0.extent(0);
+  auto sub_first_s1         = KE::begin(sub_static_view_1d_1);  // 1, 3, 5, ...
+  auto sub_last_s1          = sub_first_s1 + sub_static_view_1d_1.extent(0);
+
+  KE::Impl::expect_no_overlap(sub_first_s0, sub_last_s0, sub_first_s1,
+                              sub_last_s1);
+
+  Kokkos::View<value_type**, Kokkos::LayoutLeft> dynamic_view_2d{
+      "std-algo-test-2d-contiguous-view-dynamic", 2, extent0};
+  auto sub_dynamic_view_1d_0 = Kokkos::subview(dynamic_view_2d, 0, Kokkos::ALL);
+  auto sub_dynamic_view_1d_1 = Kokkos::subview(dynamic_view_2d, 1, Kokkos::ALL);
+  auto sub_first_d0 = KE::begin(sub_dynamic_view_1d_0);  // 0, 2, 4, ...
+  auto sub_last_d0  = sub_first_d0 + sub_dynamic_view_1d_0.extent(0);
+  auto sub_first_d1 = KE::begin(sub_dynamic_view_1d_1);  // 1, 3, 5, ...
+  auto sub_last_d1  = sub_first_d1 + sub_dynamic_view_1d_1.extent(0);
+
+  KE::Impl::expect_no_overlap(sub_first_d0, sub_last_d0, sub_first_d1,
+                              sub_last_d1);
+
+  Kokkos::LayoutStride layout2d{2, 3, extent0, 2};
+  Kokkos::View<value_type**, Kokkos::LayoutStride> strided_view_2d{
+      "std-algo-test-2d-contiguous-view-strided", layout2d};
+  auto sub_strided_view_1d_0 = Kokkos::subview(strided_view_2d, 0, Kokkos::ALL);
+  auto sub_strided_view_1d_1 = Kokkos::subview(strided_view_2d, 1, Kokkos::ALL);
+  auto sub_first_st0 = KE::begin(sub_strided_view_1d_0);  // 0, 6, 12, ...
+  auto sub_last_st0  = sub_first_st0 + sub_strided_view_1d_0.extent(0);
+  auto sub_first_st1 = KE::begin(sub_strided_view_1d_1);  // 1, 7, 13, ...
+  auto sub_last_st1  = sub_first_st1 + sub_strided_view_1d_1.extent(0);
+
+  KE::Impl::expect_no_overlap(sub_first_st0, sub_last_st0, sub_first_st1,
+                              sub_last_st1);
```

Comment:
The non-overlapping case could be run even when the preconditions are not checked.
You could wrapt the check in `TEST_NO_THROW()`.
That way you don't need the dummy passing assertion below.

Reply:
For these cases, checks are made to confirm no-overlapping. 
Sure, I will wrap them with `TEST_NO_THROW` and remove the last `EXPECT_TRUE(true)`